packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -278,7 +278,7 @@ contract Bridge is EssentialContract, IBridge {
                 // use remaining gas
                 uint256 gasLimit = msg.sender == _message.destOwner ? gasleft() : _message.gasLimit;
 
-                if (_invokeMessageCall(_message, msgHash, gasLimit)) {
+                if (gasLimit != 0 && _invokeMessageCall(_message, msgHash, gasLimit)) {
```

Comment:
I think it is not necessary, because in `_invokeMessageCall()` we already revert if gasLimit is actually 0.